### ISSUE TITLE
"Unable to update ipcache map entry on pod add" warning

### ISSUE THREAD

**Issue by christarazi** [cilium/cilium#17223]
## Bug report

```
level=warning msg="Unable to update ipcache map entry on pod add" error="ipcache entry for podIP 10.244.1.78 owned by kvstore or agent" k8sNamespace=default k8sPodName=pod-to-b-intra-node-hostport-76b9ddd5dc-8pkwq new-hostIP=10.244.1.78 new-podIP=10.244.1.78 new-podIPs="[{10.244.1.78}]" old-hostIP=10.244.1.78 old-podIP=10.244.1.78 old-podIPs="[{10.244.1.78}]" subsys=k8s-watcher
```

The warning so far seems to be benign. It doesn't seem to correlate with connectivity issues. 

Likely the commit introducing this "regression":
```
❯ glog v1.9.6..1.9.7 | rg ipcache
* a661ececa6 ipcache: Expose source when dumping entries
```

a661ececa6

**General Information**

- Observed in v1.9.7+ (does not occur in v1.9.6)

**How to reproduce the issue**

1. Deploy Cilium v1.9.7 or later
2. Deploy some pods
3. Kill pods to cause them to be recreated
4. The warnings should show up as the pods are recreated

**Comment by aanm** [comment]
@encodeflush this has been fixed in Cilium 1.10 and 1.11. Can you please check?

**Comment by aanm** [comment]
Closing as this as been fixed in 1.10 and 1.11. Feel free to open a new issue if you still see it happening in those versions.
